tools/wptrunner/wptrunner/instruments.py

```diff
@@ -0,0 +1,108 @@
+import time
+import threading
+from Queue import Queue
+
+"""Instrumentation for measuring high-level time spent on various tasks inside the runner.
+
+This is lower fidelity than an actual profile, but allows custom data to be considered,
+so that we can see the time spent in specific tests and test directories
+
+
+Instruments are intended to be used as context managers with the return value of __enter__
+containing the user-facing API e.g.
+
+with Instrument(*args) as recording:
+    recording.set(["init"])
+    do_init()
+    recording.pause()
+    for thread in test_threads:
+       thread.start(recording, *args)
+    for thread in test_threads:
+       thread.join()
+    recording.set(["teardown"])
```

Comment:
```suggestion
    recording.set(["teardown"])   # un-pauses the Instrument
```

Assuming this is true, this might make it clearer to users of the API.